source/parse.h

```diff
@@ -4707,6 +4722,11 @@ class parser
     )
         -> std::unique_ptr<statement_node>
     {
+        if (!done() && curr().type() == lexeme::Semicolon) {
+            error("empty statement is not allowed - remove extra semicolon");
```

Comment:
I think this works, but just a note: Emitting an error here (not just returning null) could interfere with productions where *statement* is one alternative but where if it doesn't parse as a statement we would want to try other fallback production(s) instead. I don't think that actually happens today in the grammar though, so this should be okay.

Reply:
Basically, the scenario would be "is there any production where a statement could appear, where if it's not a statement something else that starts with a `;` could appear instead" -- I think there isn't today